### ISSUE TITLE
Error when importing database from different package

### ISSUE THREAD

**Issue by petertoth** [vuex-orm/vuex-orm#388]
Hi,

I have a lerna monorepo and I'm putting all model logic into one package.

Then I've got a nuxt app in another package where I'm trying to install an imported database like this:

```js
// foo-nuxt/plugins/vuex-orm.js

import VuexORM from '@vuex-orm/core' 
import database from 'foo-models/client/database'

export default ({ store }) => {
  VuexORM.install(database)(store)
}
```

The database file looks like this:

```js
// foo-models/client/database.js

import { Database } from '@vuex-orm/core'
import Website from './models/website'
import Page from './models/page'

const database = new Database()

database.register(Website)
database.register(Page)

export default database
```

When I start my nuxt app, I get the following error:

```bash
vendors.app.js:25306 error TypeError: Cannot read property 'models' of undefined
    at new Schema (vendors.app.js:7929)
    at Function.push.../foo-models/node_modules/@vuex-orm/core/dist/vuex-orm.esm.js.Schema.create (vendors.app.js:7936)
    at vendors.app.js:8110
    at Array.forEach (<anonymous>)
    at Database.push.../foo-models/node_modules/@vuex-orm/core/dist/vuex-orm.esm.js.Database.createSchema (vendors.app.js:8109)
    at Database.push.../foo-models/node_modules/@vuex-orm/core/dist/vuex-orm.esm.js.Database.start (vendors.app.js:8052)
    at push../node_modules/@vuex-orm/core/dist/vuex-orm.esm.js.install (vendors.app.js:18949)
    at app.js:11496
    at _callee2$ (app.js:7516)
    at tryCatch (commons.app.js:6491)
```

I did a little bit of digging and it turns out that there are two `Container` singletons so when the database is installed: https://github.com/vuex-orm/vuex-orm/blob/ba2da6d81d515f362c3554fdd5a4866c4234f2a9/src/store/install.ts#L13-L18
the `Container` contains the installed database but when accessing database from models:
https://github.com/vuex-orm/vuex-orm/blob/ba2da6d81d515f362c3554fdd5a4866c4234f2a9/src/model/Model.ts#L263-L265

This function returns `undefined`. The error itself is in `Schema` where `database()` returns `undefined`.
https://github.com/vuex-orm/vuex-orm/blob/ba2da6d81d515f362c3554fdd5a4866c4234f2a9/src/schema/Schema.ts#L22-L28

May I ask whether this is the wrong approach on my end or a bug within Vuex ORM?

Thank you

**Comment by kiaking** [comment]
Hi @petertoth. Thank you so much for the report! This is wired... The code looks OK.

I'm not sure why there'll be 2 Containers. Is it possible for you to create reproduction code for this issue? I've never tried mono repo project with Vuex ORM, so maybe Vuex ORM missing something here.

**Comment by kiaking** [comment]
@petertoth Hi! Do you have any update on this issue?

**Comment by petertoth** [comment]
Hi, I wasn't able to set up a monorepo in Codesandbox and then got somehow busy. Will try to publish a reproduction repo later this week!

**Comment by kiaking** [comment]
@petertoth Thank you for the response! I'll be looking forward to reproduction!

**Comment by petertoth** [comment]
When creating the reproduction, I have noticed that I'm importing `VuexORM` in both of my packages and this is probably causing the problem.

So when I export the install function from my models package like this:
```js
// foo-models/client/database
import VuexORM, { Database } from '@vuex-orm/core'
import Post from './models/post'

const database = new Database()

database.register(Post)

export default (store) => {
  VuexORM.install(database)(store)
}
```

and then use it in my nuxt app:

```js
// foo-nuxt/plugins/vuex-orm
import installDatabase from 'app-models/client/database'

export default ({ store }) => {
  installDatabase(store)
}
```

Everything works just fine. I think you can close the issue and thanks for the support and creating this great project!

**Comment by kiaking** [comment]
@petertoth Ah, that makes sense! Thank you so much for the confirmation! Please feel free to open another issue if there's anything! 🤝

**Comment by mmintel** [comment]
Hi there,

I am doing like the almost the same thing:

```
export default function ({ store, axios, models = [] } = {}) {
  // Initialize new VuexORM database
  const database = new VuexORM.Database();

  // Merge default and custom models
  let allModels = [ ...Object.keys(defaultModels).map(key => defaultModels[key]) ];
  if (models && models.length) {
    allModels = [ ...allModels, ...models ];
  }

  console.log('REGISTERING', allModels);

  // Register all models to database
  allModels.forEach((model) => { database.register(model); });

  console.log(database);
  VuexORM.install(database)(store);

  // Install axios
  if (axios) {
    VuexORM.use(VuexORMAxios, {
      database,
      http: {
        axios,
      },
    });
  }
}
```

using like

```
import { createStore } from 'lib';

export default async ({ store, app }) => {
  createStore({
    axios: app.$axios,
    models: require('~/models').default,
    store,
  })
};
```

logs:
```
    REGISTERING [
      [Function: e] { entity: 'branch', methodConf: { http: [Object] } },
      [Function: e] { entity: 'resource', methodConf: { http: [Object] } },
      [Function: e] {
        entity: 'resource_class',
        primaryKey: 'name',
        methodConf: { http: [Object] }
      },
      [Function: o] { entity: 'sample_model' }
    ]
```

and for database:
```
t {
      entities: [
        { name: 'branch', base: 'branch', model: [Function], module: {} },
        {
          name: 'resource',
          base: 'resource',
          model: [Function],
          module: {}
        },
        {
          name: 'resource_class',
          base: 'resource_class',
          model: [Function],
          module: {}
        },
        {
          name: 'sample_model',
          base: 'sample_model',
          model: [Function],
          module: {}
        }
      ],
      schemas: {}
    }
```

but running my app throws `[error] Cannot read property 'models' of undefined`

@kiaking @petertoth any advice? :D for me everything looks quite correct.

also interesting: my tests in my `lib` package where I don't use nuxt are running fine.
I have a test:
```
  it('Should install the custom models', () => {
    const store = new Vuex.Store();
    createStore({
      store,
      models: [ SampleModel ],
    });
    expect(store.state.entities).toHaveProperty('sample_model');
  });
```

and my test logs

```
    REGISTERING [
      [Function: Branch] {
        entity: 'branch',
        methodConf: { http: [Object] }
      },
      [Function: Resource] {
        entity: 'resource',
        methodConf: { http: [Object] }
      },
      [Function: ResourceClass] {
        entity: 'resource_class',
        primaryKey: 'name',
        methodConf: { http: [Object] }
      },
      [Function: SampleModel] { entity: 'sample_model' }
    ]
```

and

```
Database {
      entities: [
        { name: 'branch', base: 'branch', model: [Function], module: {} },
        {
          name: 'resource',
          base: 'resource',
          model: [Function],
          module: {}
        },
        {
          name: 'resource_class',
          base: 'resource_class',
          model: [Function],
          module: {}
        },
        {
          name: 'sample_model',
          base: 'sample_model',
          model: [Function],
          module: {}
        }
      ],
      schemas: {}
    }
```

**Comment by petertoth** [comment]
Hey @mmintel, I don't see anything wrong with your implementation. Just make sure you're importing VuexORM in your whole system only once when you're using this implementation. I've also seen similar solution to this problem which was declaring VuexORM as external dependency in Webpack - provided you are using Webpack.

Also, there's already a pull request opened which is supposed to fix this problem: https://github.com/vuex-orm/vuex-orm/pull/424

Hope that helps :)

**Comment by mmintel** [comment]
Okay that's funny but it helped, thanks!

I was adding models in my library which used VuexORM and added other models in my project using `import { Model } from '@vuex-orm/core'`. Now my library exports Model from Vuex-ORM and my project uses the Model exposed from my library and doesn't access VuexORM directly. That works. But this is very weird behaviour.

**Comment by petertoth** [comment]
Sweet! I'm including a [link](https://vuex-orm.slack.com/archives/CA0AXG66L/p1566290282063100?thread_ts=1566083817.062100&cid=CA0AXG66L) to a described solution from other user in case you were wondering how to solve this more elegantly. Cheers!
